<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SeerDebugDialogForm</class>
 <widget class="QDialog" name="SeerDebugDialogForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>550</width>
    <height>661</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Select Executable to Debug</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <widget class="QGroupBox" name="executableNameGroupBox">
     <property name="title">
      <string>Executable Name</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLineEdit" name="executableNameLineEdit">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>The path and name of an executable to debug.</string>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QToolButton" name="executableNameToolButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Open a dialog to select an executable.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../SeeDataStructureRc.qrc">
          <normaloff>:/resources/RelaxLightIcons/document-open.svg</normaloff>:/resources/RelaxLightIcons/document-open.svg</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="launchMethodGroupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>100</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Launch Method</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QGridLayout" name="launchMethodGridLayout">
        <item row="2" column="1" colspan="3">
         <widget class="QLineEdit" name="loadBreakpointsFilenameLineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>100</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>A file containing previously saved breakpoints.</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="10" column="0">
         <widget class="QLabel" name="connectProgramSerialLabel">
          <property name="text">
           <string>Serial</string>
          </property>
         </widget>
        </item>
        <item row="10" column="3" colspan="2">
         <spacer name="horizontalSpacer_1">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>298</width>
            <height>22</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="attachProgramPidLabel">
          <property name="text">
           <string>Process PID</string>
          </property>
         </widget>
        </item>
        <item row="11" column="1">
         <widget class="QComboBox" name="connectProgramParityComboBox">
          <property name="toolTip">
           <string>Serial port parity setting.</string>
          </property>
          <item>
           <property name="text">
            <string>none</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>odd</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>even</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="7" column="0" colspan="5">
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="9" column="0">
         <widget class="QLabel" name="connectProgramHostPortLabel">
          <property name="text">
           <string>Gdbserver</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0" colspan="2">
         <widget class="QRadioButton" name="attachProgramRadioButton">
          <property name="toolTip">
           <string>Attach to an already running program.</string>
          </property>
          <property name="text">
           <string>Attach to a local program</string>
          </property>
         </widget>
        </item>
        <item row="13" column="0" colspan="2">
         <widget class="QRadioButton" name="loadCoreRadioButton">
          <property name="toolTip">
           <string>Load a core file to debug.</string>
          </property>
          <property name="text">
           <string>Load a core file</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QCheckBox" name="runProgramBreakInMainCheckBox">
          <property name="toolTip">
           <string>Set a breakpoint in main() before running program.</string>
          </property>
          <property name="text">
           <string>Break in main()</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <property name="tristate">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="12" column="0" colspan="5">
         <widget class="Line" name="line_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QLineEdit" name="attachProgramPidLineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>100</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>The process pid.</string>
          </property>
          <property name="inputMask">
           <string>999999</string>
          </property>
          <property name="placeholderText">
           <string>pid</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="runProgramArgumentsLabel">
          <property name="text">
           <string>Arguments</string>
          </property>
         </widget>
        </item>
        <item row="10" column="1">
         <widget class="QLineEdit" name="connectProgramBaudLineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>100</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Serial port baud rate.</string>
          </property>
          <property name="inputMask">
           <string>999999</string>
          </property>
          <property name="placeholderText">
           <string>baudrate</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="loadBreakpointsFilenameLabel">
          <property name="text">
           <string>Breakpoints</string>
          </property>
         </widget>
        </item>
        <item row="11" column="0">
         <widget class="QLabel" name="connectProgramParityLabel">
          <property name="text">
           <string>Parity</string>
          </property>
         </widget>
        </item>
        <item row="6" column="3" colspan="2">
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>298</width>
            <height>14</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QRadioButton" name="runProgramRadioButton">
          <property name="toolTip">
           <string>Run the program.</string>
          </property>
          <property name="text">
           <string>Run a program</string>
          </property>
         </widget>
        </item>
        <item row="6" column="2">
         <widget class="QToolButton" name="attachProgramPidToolButton">
          <property name="toolTip">
           <string>Open a dialog to select a pid on the local system.</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
        <item row="14" column="0">
         <widget class="QLabel" name="loadCoreFilenameLabel">
          <property name="text">
           <string>Core</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0" colspan="2">
         <widget class="QRadioButton" name="connectProgramRadioButton">
          <property name="toolTip">
           <string>Connect to an already running program started with gdbserver.</string>
          </property>
          <property name="text">
           <string>Connect to a gdbserver running a program</string>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QToolButton" name="loadBreakpointsFilenameToolButton">
          <property name="toolTip">
           <string>Open a dialog to select a breakpoints file.</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../SeeDataStructureRc.qrc">
            <normaloff>:/resources/RelaxLightIcons/document-open.svg</normaloff>:/resources/RelaxLightIcons/document-open.svg</iconset>
          </property>
         </widget>
        </item>
        <item row="9" column="1" colspan="4">
         <widget class="QLineEdit" name="connectProgramHostPortLineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>100</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>How to access the remote gdbserver.</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>host:port or /dev/serialport</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="5">
         <widget class="Line" name="line_1">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="4">
         <widget class="QLineEdit" name="runProgramArgumentsLineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>100</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>The arguments to pass to the program.</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="14" column="1" colspan="3">
         <widget class="QLineEdit" name="loadCoreFilenameLineEdit">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>100</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>The path and name of a core file.</string>
          </property>
          <property name="clearButtonEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="14" column="4">
         <widget class="QToolButton" name="loadCoreFilenameToolButton">
          <property name="toolTip">
           <string>Open a dialog to select a core file.</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../SeeDataStructureRc.qrc">
            <normaloff>:/resources/RelaxLightIcons/document-open.svg</normaloff>:/resources/RelaxLightIcons/document-open.svg</iconset>
          </property>
         </widget>
        </item>
        <item row="11" column="3" colspan="2">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>298</width>
            <height>19</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="executableWorkingDirectoryGroupBox">
     <property name="toolTip">
      <string/>
     </property>
     <property name="title">
      <string>Working Directory</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QToolButton" name="executableWorkingDirectoryToolButton">
        <property name="toolTip">
         <string>Open a dialog to select a path.</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../SeeDataStructureRc.qrc">
          <normaloff>:/resources/RelaxLightIcons/document-open.svg</normaloff>:/resources/RelaxLightIcons/document-open.svg</iconset>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLineEdit" name="executableWorkingDirectoryLineEdit">
        <property name="toolTip">
         <string>The working directory path to tell GDB.</string>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
  <zorder>launchMethodGroupBox</zorder>
  <zorder>executableNameGroupBox</zorder>
  <zorder>executableWorkingDirectoryGroupBox</zorder>
  <zorder>buttonBox</zorder>
 </widget>
 <resources>
  <include location="../SeeDataStructureRc.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SeerDebugDialogForm</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SeerDebugDialogForm</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
